measure.go

```diff
@@ -145,6 +172,45 @@ func (m *measure) Query(q query.Query) (query.Results, error) {
 	return res, err
 }
 
+func (m *measure) Check() error {
+	defer recordLatency(m.checkLatency, time.Now())
+	m.checkNum.Inc()
+	if c, ok := m.backend.(datastore.CheckedDatastore); ok {
+		err := c.Check()
+		if err != nil {
+			m.checkErr.Inc()
+		}
+		return err
+	}
+	return nil
+}
+
+func (m *measure) Scrub() error {
+	defer recordLatency(m.scrubLatency, time.Now())
+	m.scrubNum.Inc()
+	if c, ok := m.backend.(datastore.ScrubbedDatastore); ok {
+		err := c.Scrub()
+		if err != nil {
+			m.scrubErr.Inc()
+		}
+		return err
+	}
+	return nil
+}
+
+func (m *measure) CollectGarbage() error {
+	defer recordLatency(m.gcLatency, time.Now())
+	m.gcNum.Inc()
```

Comment:
Shouldn't we only modify the metrics if we actually do a GC? That is, shouldn't these go inside the condition?

Reply:
The count metrics are described as `Total number of Datastore.[..] calls`, so my reasoning was that we want to count number of method invocations, not what they are actually doing.

Reply:
I see... Meh.